src/lib/translators/mongo.ts

```diff
@@ -545,11 +548,22 @@ const mapper: Partial<StepMatcher<MongoStep>> = {
     },
   }),
   filter: filterstepToMatchstep,
-  formula: (step: Readonly<S.FormulaStep>) => ({
-    $addFields: {
-      [step.new_column]: buildMongoFormulaTree(math.parse(step.formula)),
-    },
-  }),
+  formula: (step: Readonly<S.FormulaStep>) => {
+    // replace all `[column name]` values by a pseudo and store it in `pseudos`
+    // `buildMongoFormulaTree` will re-put the name back
+    const pseudos: { [key: string]: string } = {};
+    const formulaPseudotised = step.formula.replace(/\[.*?\]/g, columName => {
+      const pseudo = `column_n_${Object.keys(pseudos).length}`;
```

Comment:
We have a "tradition" of prefixing such vqb-specific variables with `vqb`, cf. https://github.com/ToucanToco/weaverbird/blob/master/src/lib/translators/mongo.ts#L188 for instance. Maybe we could do that here?